KosherCocoa/Library/Core/Calendar/Chagim/KCJewishCalendar.m

```diff
@@ -36,129 +36,136 @@ - (NSInteger)yomTovIndex
+            else if (day == 15 || day == 21
+                     || (!inIsrael && (day == 16 || day == 22)))
             {
                 return kPesach;
             }
-            else if (([self currentHebrewDayOfMonth] >= 17 && [self currentHebrewDayOfMonth] <= 20)
-                     || ([self currentHebrewDayOfMonth] == 16 && inIsrael))
+            else if ((day >= 17 && day <= 20)
+                     || (day == 16 && inIsrael))
             {
                 return kCholHamoedPesach;
             }
             if (useModernHolidays
-                && (([self currentHebrewDayOfMonth] == 26 && [self currentDayOfTheWeek] == 5)
-                    || ([self currentHebrewDayOfMonth] == 28 && [self currentDayOfTheWeek] == 1)
-                    || ([self currentHebrewDayOfMonth] == 27 && [self currentDayOfTheWeek] == 3)|| ([self currentHebrewDayOfMonth] == 27 && [self currentDayOfTheWeek] == 5)))
+                && ((day == 26 && dayOfWeek == 5)
+                    || (day == 28 && dayOfWeek == 2)
+                    || (day == 27 && dayOfWeek != 1 && dayOfWeek != 6)))
             {
                 return kYomHashoah;
             }
             break;
         case KCHebrewMonthIyar:
             if (useModernHolidays
-                && (([self currentHebrewDayOfMonth] == 4 && [self currentDayOfTheWeek] == 3)
-                    || (([self currentHebrewDayOfMonth] == 3 || [self currentHebrewDayOfMonth] == 2)&& [self currentDayOfTheWeek] == 4)|| ([self currentHebrewDayOfMonth] == 5 && [self currentDayOfTheWeek] == 2)))
+                && ((day == 4 && dayOfWeek == 3)
+                    || ((day == 3 || day == 2) && dayOfWeek == 4) || (day == 5 && dayOfWeek == 2)))
             {
                     return kYomHazikaron;
                 }
             // if 5 Iyar falls on Wed Yom Haatzmaut is that day. If it falls on Friday or Shabbos it is moved back to
             // Thursday. If it falls on Monday it is moved to Tuesday
             if (useModernHolidays
-                && (([self currentHebrewDayOfMonth] == 5 && [self currentDayOfTheWeek] == 4)
-                    || (([self currentHebrewDayOfMonth] == 4 || [self currentHebrewDayOfMonth] == 3)&& [self currentDayOfTheWeek] == 5)|| ([self currentHebrewDayOfMonth] == 6 && [self currentDayOfTheWeek] == 3)))
+                && ((day == 5 && dayOfWeek == 4)
+                    || ((day == 4 || day == 3) && dayOfWeek == 5) || (day == 6 && dayOfWeek == 3)))
             {
                     return kYomHaatzmaut;
                 }
-            if ([self currentHebrewDayOfMonth] == 14)
+            if (day == 14)
             {
                 return kPesachSheni;
             }
-            if (useModernHolidays && [self currentHebrewDayOfMonth] == 28)
+            if (day == 18)
             {
-                return kYomHashoah;
+                return kLagBaomer;
```

Comment:
added lag baomer

Reply:
This was also where it would return Yom Hashoah accidentally

Reply:
https://github.com/MosheBerman/KosherCocoa/pull/149/files#diff-8207d5da44d8a10dbfce0c57acc32376d5d71254c789a4f1b2cf04769546c14dR64